File: pythonSocket/echoClient.py
# import socket module
import socket 
 
# creating socket client
client_socket = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
 
# connect to server in defined address and port
client_socket.connect(('localhost', 5000))
 
# send message to server
client_socket.send("Hi server ... ")
 
# receive message from server, 1024 is buffer size in bytes
message = client_socket.recv(1024)
 
# print message
print "From server: " + message
 
# close socket client 
client_socket.close()

File: javaSocket/echoClient/src/echoclient/EchoClient.java
/*
 * To change this template, choose Tools | Templates
 * and open the template in the editor.
 */
package echoclient;

/**
 *
 * @author Hudan
 */

import java.net.*;
import java.io.*;

public class EchoClient {

    /**
     * @param args the command line arguments
     */
    public static void main(String[] args) {
        // TODO code application logic here
        try {
            /*
             * variable initialization
             */
            Socket clientSocket;
            DataInputStream inputStream;
            DataOutputStream outputStream;
            String message;
            
            /*
             * create socket, preapring data output stream, send to server
             */
            clientSocket = new Socket("localhost", 5000);
            outputStream = new DataOutputStream(clientSocket.getOutputStream());
            outputStream.writeUTF("Message from client");
            
            /*
             * preparing data input stream, receive message from server, print
             */
            inputStream = new DataInputStream(clientSocket.getInputStream());
            message = inputStream.readUTF();
            System.out.println("From server: " + message);
            
            /*
             * close data output stream, data input stream, and client socket
             */
            outputStream.close();
            inputStream.close();
            clientSocket.close();
        }
        catch(IOException e) {
            System.out.println("IO exception: " + e.getMessage());
        }
    }
}
